obal/data/roles/setup-sources/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- set_fact:
+    pkg_dir: "{{ inventory_dir }}/packages/{{ inventory_hostname }}"
+    annex_opts: annex_opts
+
+- name: get spec file
+  find:
+    pattern: "*.spec"
+    path: "{{ pkg_dir }}"
+  register: spec_file
+  tags:
+    - always
+
+- set_fact:
+    spec_file_path: "{{ spec_file.files[0].path }}"
+  tags:
+    - always
+
+- name: 'Extract Source0 from spec'
+  shell: "spectool --list-files --source 0 {{ spec_file_path }} | awk '{print $2}'"
+  register: source0
+  changed_when: False
+  tags:
+    - always
+
+- set_fact:
+    source0: "{{ source0.stdout }}"
+  tags:
+    - always
+
+- name: 'Extract sourcebase from Source0'
+  shell: "basename {{ source0 }}"
+  register: sourcebase
+  changed_when: False
+  tags:
+    - always
```

Comment:
I think you can combine this with the `set_fact` above using the `basename` filter Ansible provides